runtime/compiler/z/codegen/J9S390CHelperLinkage.cpp

```diff
@@ -266,14 +266,19 @@ class RealRegisterManager
  *    \param deps The pre register dependency conditions that will be filled by this function to attach within ICF
  *    \param returnReg TR::Register* allocated by consumer of this API to hold the result of the helper call,
  *           If passed, this function uses it to store return value from helper instead of allocating new register
+ *    \param helperIndex The helper function's index
  *    \return TR::Register *helperReturnResult, gets the return value of helper function and return to the evaluator. 
  */
-TR::Register * TR::S390CHelperLinkage::buildDirectDispatch(TR::Node * callNode, TR::RegisterDependencyConditions **deps, TR::Register *returnReg)
+TR::Register * TR::S390CHelperLinkage::buildDirectDispatch(TR::Node * callNode,
+                                                           TR::RegisterDependencyConditions **deps,
+                                                           TR::Register *returnReg,
+                                                           bool forceFastPath,
+                                                           TR_RuntimeHelper helperIndex)
    {
    RealRegisterManager RealRegisters(cg());
    bool isHelperCallWithinICF = deps != NULL;
-   // TODO: Currently only jitInstanceOf is fast path helper. Need to modify following condition if we add support for other fast path only helpers
-   bool isFastPathOnly = callNode->getOpCodeValue() == TR::instanceof;
```

Comment:
Could we leave this TODO around? This is definitely something we still want to do once we refactor the runtime helpers into a sane table with properties.

Reply:
TODO is added back